PR of Experimental Branch

steevaavoo opened the pull request:
Fixed pester tests (thank you Adam!), removed redundant help uris, removed unnecessary parameter sets.

steevaavoo committed:
Deleted unnecessary parameter set names

steevaavoo committed:
Deleted typo

steevaavoo committed:
Removed helpuris
Added scriptmethod

steevaavoo committed:
Fixed tests

adamrushuk commented on ManageRDSessions/ManageRDSessions.psm1:
```diff
@@ -46,24 +46,23 @@ function Get-sbRDSession {
     param(
-        [cmdletbinding(DefaultParameterSetName = 'State')]
-        [Parameter(Mandatory = $false, ParameterSetName = 'State')]
+        [cmdletbinding()]
```
casing should be `CmdletBinding`

steevaavoo replied:
Corrected throughout

adamrushuk commented on ManageRDSessions/ManageRDSessions.psm1:
```diff
@@ -46,24 +46,23 @@ function Get-sbRDSession {
-        [Parameter(Mandatory = $false, ParameterSetName = 'State')]
+        [Parameter(Mandatory = $false)]
         [Alias("Self")]
         [switch]$IncludeSelf = $false,
```
`$false` is the default for `[Switch]` anyway, so this is superfluous; like your face!

steevaavoo replied:
Oooooooooooooo

steevaavoo replied:
$false default removed

adamrushuk commented on ManageRDSessions/ManageRDSessions.psm1:
```diff
@@ -90,43 +89,60 @@ function Get-sbRDSession {
+                Write-Verbose "Querying RD Session Collection for [$SessionState] sessions - including [$env:USERNAME]"
+                $sessions = Get-RDUserSession | Where-Object {
+                    $_.SessionState -like $stateLookup.$SessionState`
+                        -and ( $_.IdleTime / 60000 ) -ge $MinimumIdleMins`
```
Only indent one level when lines run over

steevaavoo replied:
See below backtick-related comment

adamrushuk commented on ManageRDSessions/ManageRDSessions.psm1:
```diff
@@ -90,43 +89,60 @@ function Get-sbRDSession {
+            }
+            else {
+                Write-Verbose "Querying RD Session Collection for [$SessionState] sessions"
+                $sessions = Get-RDUserSession | Where-Object {
```
Only indent one level, like this:
```     
$sessions = Get-RDUserSession | Where-Object {
    $_.SessionState -like $stateLookup.$SessionState
    -and ( $_.IdleTime / 60000 ) -ge $MinimumIdleMins
    -and $_.UserName -ne "$env:USERNAME"
}           
```

adamrushuk replied:
I think you can remove the backticks too

steevaavoo replied:
Done - it allowed the formatter to correct the indentation as predicted.

adamrushuk commented on ManageRDSessions/ManageRDSessions.psm1:
```diff
@@ -90,43 +89,60 @@ function Get-sbRDSession {
-End {
-    #Intentionally empty
-}
+    End {
```
remove the End block if unused

steevaavoo replied:
Done - also with unnecessary begin blocks throughout

adamrushuk commented on ManageRDSessions/ManageRDSessions.psm1:
```diff
@@ -90,43 +89,60 @@ function Get-sbRDSession {
 
+# Adding a Script Method to Get-sbRDSession to allow sending messages to the session(s) found.
+$myType = "Custom.SB.RDSession"
+Update-TypeData -TypeName $myType -MemberType ScriptMethod -MemberName SendMessage -Value {
```
Mind blown!

adamrushuk commented on tests/Get-sbRDSession.Tests.ps1:
```diff
@@ -27,5 +27,17 @@ InModuleScope $ModuleName {
         It "only accepts active, idle, disconnected, connected or any" {
             { Get-sbRDSession -SessionState Dancing } | Should Throw
```
You can validate the scenarios that should pass too, like:
```
{ Get-sbRDSession -SessionState Active} | Should Not Throw
```
I've not tested the above, but you get the gist.

steevaavoo replied:
Noted and added.

adamrushuk reviewed (changes requested):
Amazing work my friend; you've come far!

steevaavoo committed:
Changed requested items from PR
Removed superfluous begin/end blocks
Removed superfluous backticks
Changed CmdletBinding casing

steevaavoo committed:
Added should not throw to tests

steevaavoo commented:
So I've made the requested changes, and once again noticed the formatter's propensity to get "funky" with the closing }s - can you see the same in my latest commit?

steevaavoo commented:
Aside from that, are we looking at a release-ready module?

adamrushuk commented on ManageRDSessions/ManageRDSessions.psm1:
```diff
@@ -46,24 +46,23 @@ function Get-sbRDSession {
-        [Parameter(Mandatory = $false, ParameterSetName = 'State')]
+        [Parameter(Mandatory = $false)]
         [Alias("Self")]
         [switch]$IncludeSelf = $false,
```
This should just be `[switch]$IncludeSelf`

adamrushuk commented:
almost there

adamrushuk reviewed (changes requested):
Apart from the Switch $false still showing, it's looking ready.
I mentioned dynamically dot sourcing function files - instead of one big .psm1 (as shown here: https://github.com/adamrushuk/PSvCloud/blob/master/PSvCloud/PSvCloud.psm1) - but that will mess with diffs and show as new files, so I'd leave that to a separate branch and PR in future after a restructure.
